plutus-contract/src/Language/Plutus/Contract/Tx.hs

```diff
@@ -53,6 +57,16 @@ data UnbalancedTx = UnbalancedTx
 
 Lens.TH.makeLenses ''UnbalancedTx
 
+-- TODO: this is a bit of a hack, I'm not sure quite what the best way to avoid this is
+fromLedgerTx :: L.Tx -> UnbalancedTx
+fromLedgerTx tx = UnbalancedTx
+            { _inputs = L.txInputs tx
+            , _outputs = L.txOutputs tx
+            , _forge = L.txForge tx
+            , _validityRange = L.txValidRange tx
+            , _requiredSignatures = mempty
```

Comment:
If you put the public keys from `Ledger.Tx.txSignatures` in here then at least you will be able to generate the original `L.Tx` from the `Unbalanced` (provided you have the right keys)